instrumentation/opentelemetry-instrumentation-starlette/tests/test_starlette_instrumentation.py

```diff
@@ -667,15 +667,17 @@ async def _(websocket: WebSocket) -> None:
 
 
 class TestHTTPAppWithCustomHeaders(TestBaseWithCustomHeaders):
-    @patch.dict(
-        "os.environ",
-        {
-            OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SANITIZE_FIELDS: ".*my-secret.*",
-            OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SERVER_REQUEST: "Custom-Test-Header-1,Custom-Test-Header-2,Custom-Test-Header-3,Regex-Test-Header-.*,Regex-Invalid-Test-Header-.*,.*my-secret.*",
-            OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SERVER_RESPONSE: "Custom-Test-Header-1,Custom-Test-Header-2,Custom-Test-Header-3,my-custom-regex-header-.*,invalid-regex-header-.*,.*my-secret.*",
-        },
-    )
     def setUp(self) -> None:
+        patcher = patch.dict(
```

Comment:
Previously patch was decorating `setUp` - it was active only for the duration of `setUp`, not the entire test case. Recent starlette seems to initialize middleware lazily so the patch needs to be active during the test case for it. I switched to this pattern that activates it both for `setUp` (for older eager starlette) and the test cases (for newer lazy starlette). Note, moving the decorator to the class level would loose the former.

But it seems CI is showing failures still that I can't repro locally will check it out more